runtime/environment.go

```diff
@@ -620,8 +620,21 @@ func (e *interpreterEnvironment) newOnStatementHandler() interpreter.OnStatement
 
 	return func(inter *interpreter.Interpreter, statement ast.Statement) {
 		location := inter.Location
+		if !e.coverageReport.IsProgramInspected(location) {
+			program := inter.Program.Program
+			err := e.coverageReport.InspectProgram(location, program)
+			if err != nil {
+				// TODO: Log something
+				return
+			}
```

Comment:
`InspectProgram` always returns `nil`, so maybe just remove the error return value from `InspectProgram` and then there is no need for error handling here

Reply:
Hmm, maybe I'm missing something here, but:
```go
if r.IsLocationExcluded(location) {
	return fmt.Errorf("location is excluded from coverage: %v", location)
}
```
`InspectProgram` will return early if the `location` is excluded. Side note: one use for excluding locations, is to not generate coverage for Cadence test scripts.

I could move this check in `runtime/environment.go`:

```go
return func(inter *interpreter.Interpreter, statement ast.Statement) {
	location := inter.Location
	if !e.coverageReport.IsProgramInspected(location) {
		program := inter.Program.Program
		err := e.coverageReport.InspectProgram(location, program)
		if err != nil {
			// TODO: Log something
			return
		}
	}

	...
}
```

However, I thought that it is best if the coverage package accounts for this.

Reply:
If a location is excluded from being inspected, is that really an error? I assumed `InspectProgram` can return early in that case.

Reply:
Fair point, it's not really an error. I have updated the methods to perform a NO-OP when a location is excluded. See https://github.com/onflow/cadence/pull/2296/commits/4cc2c02e21bb7491ca708d4b75c03579a8e86c87